src/Psalm/Internal/Analyzer/Statements/Expression/Fetch/ClassConstFetchAnalyzer.php

```diff
@@ -452,6 +456,255 @@ public static function analyze(
+                if ($fq_class_name !== $context->self) {
+                    $class_constant_type = $codebase->classlikes->getClassConstantType(
+                        $fq_class_name,
+                        $stmt->name->name,
+                        \ReflectionProperty::IS_PRIVATE,
+                        $statements_analyzer
+                    );
+                }
+
+                if ($class_constant_type) {
+                    if (IssueBuffer::accepts(
+                        new InaccessibleClassConstant(
+                            'Constant ' . $const_id . ' is not visible in this context',
+                            new CodeLocation($statements_analyzer->getSource(), $stmt)
+                        ),
+                        $statements_analyzer->getSuppressedIssues()
+                    )) {
+                        // fall through
+                    }
+                } elseif ($context->check_consts) {
+                    if (IssueBuffer::accepts(
+                        new UndefinedConstant(
+                            'Constant ' . $const_id . ' is not defined',
+                            new CodeLocation($statements_analyzer->getSource(), $stmt)
+                        ),
+                        $statements_analyzer->getSuppressedIssues()
+                    )) {
+                        // fall through
+                    }
+                }
+
+                return true;
+            }
+
+            if ($context->calling_method_id) {
+                $codebase->file_reference_provider->addMethodReferenceToClassMember(
+                    $context->calling_method_id,
+                    strtolower($fq_class_name) . '::' . $stmt->name->name,
+                    false
+                );
+            }
+
+            $declaring_const_id = strtolower($fq_class_name) . '::' . $stmt->name->name;
+
+            if ($codebase->alter_code && !$moved_class) {
+                foreach ($codebase->class_constant_transforms as $original_pattern => $transformation) {
+                    if ($declaring_const_id === $original_pattern) {
+                        [$new_fq_class_name, $new_const_name] = explode('::', $transformation);
+
+                        $file_manipulations = [];
+
+                        if (strtolower($new_fq_class_name) !== strtolower($fq_class_name)) {
+                            $file_manipulations[] = new \Psalm\FileManipulation(
+                                (int) $stmt->class->getAttribute('startFilePos'),
+                                (int) $stmt->class->getAttribute('endFilePos') + 1,
+                                Type::getStringFromFQCLN(
+                                    $new_fq_class_name,
+                                    $statements_analyzer->getNamespace(),
+                                    $statements_analyzer->getAliasedClassesFlipped(),
+                                    null
+                                )
```

Comment:
Am I mistaken, or would this replace the variable with the class name, thus changing the intent of the code?

Reply:
Oh, I get it now. Thanks